### ISSUE TITLE
push and set TIMESTAMPs

### ISSUE THREAD

**Issue by Gerriko** [mobizt/Firebase-Arduino-WiFi101#2]
Hi,

I am using an MKR1000 and I'm in need of setting and pushing a timestamp to my firebase project.

I've used similar methods successfully with your ESP8266 library.

So, I am just wondering if you plan to include these functions in this library or is there some issue that prevents these functions from being implemented.

Thanks.

**Comment by mobizt** [comment]
Hi Gerriko, I update the library to v1.0.3 which supports timestamp functions as you requested already.
Please try.

**Comment by Gerriko** [comment]
Thanks for the quick response.

There was one small error which was straightforward to fix. You had
"buildRequest" as per ESP8266 while for WiFi101 you needed to use
"sendRequest". This allows the program to compile, but then the
pushTimestamp returns an error with unknown reason and the setTimestamp
method did not do anything. So something in the sendRequest might not be
set up properly.

**Comment by mobizt** [comment]
Thanks somuch and sorry for the mistake, now I fix it and push the version to 1.0.4 and remove v1.0.3 to let Arduino's Library Manager to take this version instead. Please update to version 1.0.4.

**Comment by mobizt** [comment]
Sorry again, please stay with the v 1.0.2 until I get to home and test the real device and I will inform you later (within 2-3 hours).

**Comment by Gerriko** [comment]
Ok thanks for the updates. I will wait. Thanks again.

**Comment by mobizt** [comment]
Ok the version 1.0.4 works, please update.

**Comment by Gerriko** [comment]
Thanks.

**Comment by Gerriko** [comment]
I realise you've done a fantastic job with the esp8266 and esp32 firebase
libraries, but I've very quickly hit runtime memory issues with the MKR1000
/ wifi101 version when I add in my own application.

I'm using a TFT screen and require at most 4k of RAM. There is 32k RAM
available with the MKR1000 module, so somewhere along the way 28kRAM is
used up somewhere.

In the past, I found that with GSM modules they require large arrays of
strings for AT commands and I would have followed a similar PROGMEM
approach as you've used. However, I had to reserve (allocate) memory
upfront for a string buffer and then I would use string replace to swap out
commands within my reserved string buffer etc. So I only had one string
variable name that I would resuse everywhere.

Anyhow I was browsing through your code and I think there may be scope to
change things, but I can also quickly see that this will require a good
deal of work.

So for me I think will have swap out my MKR1000 for another module like an
ESP32 for now so that I can use the ESP32 library.

**Comment by mobizt** [comment]
Some Arduino's boards design are not suitable for internet usage as they want to be especially for large payload data. The library itself ised relatively small static data to keep most of work stable and dynamic data depends on user data size. Nothing to do with current code because the program memory is limited to implement manage c code. I cannot implement all features as in esp8266 due to program memory (eprom) limit instead of data mem (ram) limit.

**Comment by Gerriko** [comment]
Sorry, no need to open the issue but FWIW, I thought to share an update as
I discovered something. Looks possible like a mkr1000 board issue regarding
core API.

If I store my 4kByte image as a global variable array in my application
code, rather than as a static const or in PROGMEM, the Firebase library
works perfectly.

Regarding the problem, I should've been a little clearer. So, if using
const or PROGMEM, what I found is all the set or get Firebase functions
returns fail and the errorReason is blank, and the program continues.

As in:

  if (!Firebase.setTimestamp(firebaseData, FBpath + "...."))
  {
    Serial.println(F("Firebase Update Error"));
    Serial.println("REASON: " + firebaseData.errorReason());
  }
